### PR TITLE
A new `Atoms` class derived from `ase`

### PR DESCRIPTION
**Pull request by sudarsan-surendralal**
This is something that has been in the works for many years but hasn't been done so far. The idea is to replace our `Atoms` class with a new one which is derived from the `ase.atoms.Atoms` class. The things we need to ensure

-  All additional functionality of the previous `Atoms` class (eg. tag system) is retained. This has to be checked using extensive unit-testing
-  Backward compatibility when reading older pyiron jobs

### CODE REVIEW

**Comment by jan-janssen** [comment]
Please use the `to_hdf` and `from_hdf` functions from https://github.com/pyiron/pyiron/blob/master/pyiron/gpaw/pyiron_ase.py#L140

**Comment by sudarsan-surendralal** [comment]
It looks like you can't set vectors for magnetic moments in ase (for NCL calculations). 
```
======================================================================

ERROR: test_write_magmoms (vasp.test_vasp.TestVasp)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/pyiron/pyiron/tests/vasp/test_vasp.py", line 215, in test_write_magmoms

    self.job.structure.set_initial_magnetic_moments(magmom_ncl)

  File "/home/travis/build/pyiron/pyiron/pyiron/atomistics/structure/atoms.py", line 2938, in set_initial_magnetic_moments

    super(Atoms, self).set_initial_magnetic_moments(magmoms=magmoms)

  File "/home/travis/miniconda/lib/python3.7/site-packages/ase/atoms.py", line 608, in set_initial_magnetic_moments

    magmoms.shape[1:])

  File "/home/travis/miniconda/lib/python3.7/site-packages/ase/atoms.py", line 475, in set_array

    (name, a.shape, b.shape))

ValueError: Array "initial_magmoms" has wrong shape (8, 3) != (8,).
```

**Comment by sudarsan-surendralal** [comment]
A almost all tests pass now. The only tests that fail are:

- quickff tests
- Examples that use gpaw

**Comment by jan-janssen** [comment]
The yaff issues seem to be related to the new cell object. Most likely they need a numpy cell as before rather than the ASE cell object. Do you know how to convert the ASE cell to a 3x3 numpy matrix?

**Comment by sudarsan-surendralal** [comment]
> 
> 
> The yaff issues seem to be related to the new cell object. Most likely they need a numpy cell as before rather than the ASE cell object. Do you know how to convert the ASE cell to a 3x3 numpy matrix?

`np.array(cell)` should work. But I'm not a 100% sure where to put this

File: pyiron/gpaw/pyiron_ase.py [hunk 1]
```diff
@@ -124,12 +124,12 @@ def structure(self, basis):
     def to_hdf(self, hdf=None, group_name=None):
         super(AseJob, self).to_hdf(hdf=hdf, group_name=group_name)
         with self.project_hdf5.open("input") as hdf_input:
-            hdf_input["structure"] = self._todict()
+            hdf_input["structure"] = self.structure
```
Comment (jan-janssen) [inline pyiron/gpaw/pyiron_ase.py:127]: This part can not work.

**Comment by jan-janssen** [comment]
At least Yaff and Quickff are not fixed.

**Comment by sudarsan-surendralal** [comment]
Finally, all the tests and notebooks run! We can consider reviewing/merging after we discuss https://github.com/pyiron/pyiron/issues/797

**Comment by jan-janssen** [comment]
We are going to merge this one on August the 7th - so please test it extensively.

**Comment by sudarsan-surendralal** [comment]
The following simulation tasks need to pass before merging

- [ ] Standard VASP and LAMMPS simulations submitted to the queue

- [ ] Complicated VASP jobs with user defined elements

- [ ] VASP jobs with magnetic tags

- [ ] Interactive DFT and LAMMPS jobs

- [ ] A complicated analysis workflow of existing jobs

(feel free to add more)

**Comment by sudarsan-surendralal** [comment]
Some tests are failing randomly (nothing to do with this PR)

```
======================================================================
ERROR: test_install (base.settings.test_install.TestInstall)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/share/miniconda/envs/pyiron/lib/python3.7/http/client.py", line 554, in _get_chunk_left
    chunk_left = self._read_next_chunk_size()
  File "/usr/share/miniconda/envs/pyiron/lib/python3.7/http/client.py", line 521, in _read_next_chunk_size
    return int(line, 16)
ValueError: invalid literal for int() with base 16: b''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/share/miniconda/envs/pyiron/lib/python3.7/http/client.py", line 586, in _readinto_chunked
    chunk_left = self._get_chunk_left()
  File "/usr/share/miniconda/envs/pyiron/lib/python3.7/http/client.py", line 556, in _get_chunk_left
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/home/vsts/work/1/s/tests/base/settings/resources'

----------------------------------------------------------------------
```

**Comment by jan-janssen** [comment]
I restarted the tests, but you can also restart them yourself.

**Comment by pmrv** [comment]
I got around to run the VASP handbook with 61abcc34 and there was some breakage but nothing related the new atoms class that wasn't fixed by `ase_to_pyiron`.

**Comment by sudarsan-surendralal** [comment]
On approval, we can now merge this!

**Pull request merged**